tests/test_utils.py

```diff
@@ -1106,41 +1107,48 @@ def test_intersect(objs, expected):
 
 
 @pytest.mark.parametrize(
-    'objs, expected',
+    'objs, error_msg',
     [
         (
             [
                 pd.Index([]),
             ],
-            True,
+            None,
         ),
         (
             [
                 pd.Index([]),
                 pd.Index([]),
             ],
-            True,
+            None,
         ),
         (
             [
                 pd.Index([], name='l'),
                 pd.Index([], name='L'),
             ],
-            False,
+            "Found different level names: ['L', 'l']",
         ),
         (
             [
                 pd.Index([]),
                 pd.MultiIndex([[]], [[]]),
             ],
-            True,
+            None,
+        ),
+        (
+            [
+                pd.Index([1, 2, 3]),
+                pd.Index([10, 20], name='l'),
+            ],
+            "Found different level names: ['None', 'l']",
```

Comment:
Maybe better
```suggestion
            "Found different level names: [None, 'l']",
```
to avoid a name clash if one name is `'None'`.

BTW, that example does not work at the moment:

```python
index1 = pd.Index([])
index2 = pd.Index([], name="None")
audformat.core.utils._assert_index_alike([index1, index2])
```
returns
```
ValueError: Levels and dtypes of all objects must match.
```
without information on the non-matching names, I guess because you would have `('None', 'None')` which match.

Reply:
> to avoid a name clash if one name is `'None'`.

Mhh, the reason I convert to string here is that in order to have deterministic output I sort the result, but therefore I had to convert `None` to a string. However, I guess you a right, we should not convert the result to a string. I'll try to find a solution for it.

Reply:
Ok, we no longer convert to string. I extended the tests with your example.

Reply:
Locally tests are passing, but here I'm getting:

```
AssertionError: Regex pattern "Found\\ different\\ level\\ names:\\ \\[None,\\ 'None'\\]" does not match "Levels and dtypes of all objects must match. Found different level names: ['None', None].".
```

So it seems that after the change sorting is no longer deterministic....

Reply:
I now replaced `set()`with `dict.fromkeys()`.

Reply:
perfect